### ISSUE TITLE
Bridge.sln does not build with Visual Studio 2019 16.4.3

### ISSUE THREAD

**Issue by ChrML** [bridgedotnet/Bridge#4112]
Bridge.NET does not seem to fully build on the latest version of Visual Studio. It seems to be able to build _bridge.exe_, but executing it raises the same error in runtime.

There seems to be a dependency to `System.String.GetPathsOfAllDirectoriesAbove` that cannot be resolved.

I could be doing somehing wrong. If I am, the documentation should probably be updated, as I did exactly what's written here:
https://github.com/bridgedotnet/Bridge/wiki/Build-From-Source

### Steps To Reproduce

1. Clone the master branch of the Bridge.NET repo.
2. Open Bridge.sln in Visual Studio 2019 version 16.4.3.
3. Try to build it without changing anything.

```
1>------ Build started: Project: Bridge.Translator, Configuration: Debug Any CPU ------
1>  All packages listed in packages.config are already installed.
1>  Bridge.Translator -> D:\Git\Bridge\Compiler\Translator\bin\Debug\Bridge.Translator.dll
2>------ Build started: Project: Bridge.PostBuild, Configuration: Debug Any CPU ------
2>  Executing Bridge.Builder.Console...
2>EXEC : Bridge.NET Compiler error : Microsoft.Build.Exceptions.InvalidProjectFileException: The expression ""Attributes\ConventionAttribute.cs".GetPathsOfAllDirectoriesAbove()" cannot be evaluated. Finner ikke metoden System.String.GetPathsOfAllDirectoriesAbove.  C:\Program Files (x86)\Microsoft Visual Studio\2019\MSBuild\Current\Bin\Roslyn\Microsoft.Managed.Core.targets
2>     ved Microsoft.Build.Shared.ProjectErrorUtilities.ThrowInvalidProject(String errorSubCategoryResourceName, IElementLocation elementLocation, String resourceName, Object[] args) i /_/src/Shared/ProjectErrorUtilities.cs:linje 412
2>     ved Microsoft.Build.Shared.ProjectErrorUtilities.ThrowInvalidProject[T1,T2](IElementLocation elementLocation, String resourceName, T1 arg0, T2 arg1) i /_/src/Shared/ProjectErrorUtilities.cs:linje 98
2>     ved Microsoft.Build.Evaluation.Expander`2.Function`1.Execute(Object objectInstance, IPropertyProvider`1 properties, ExpanderOptions options, IElementLocation elementLocation) i /_/src/Build/Evaluation/Expander.cs:linje 3312
2>     ved Microsoft.Build.Evaluation.Expander`2.ItemExpander.IntrinsicItemFunctions`1.<ExecuteStringFunction>d__13.MoveNext() i /_/src/Build/Evaluation/Expander.cs:linje 2446
2>     ved Microsoft.Build.Evaluation.Expander`2.ItemExpander.<Transform>d__0`1.MoveNext() i /_/src/Build/Evaluation/Expander.cs:linje 1628
2>     ved Microsoft.Build.Evaluation.Expander`2.ItemExpander.<Transform>d__0`1.MoveNext() i /_/src/Build/Evaluation/Expander.cs:linje 1619
2>     ved Microsoft.Build.Evaluation.Expander`2.ItemExpander.ExpandExpressionCapture[S](Expander`2 expander, ItemExpressionCapture expressionCapture, IItemProvider`1 evaluatedItems, IElementLocation elementLocation, ExpanderOptions options, Boolean includeNullEntries, Boolean& isTransformExpression, List`1& itemsFromCapture) i /_/src/Build/Evaluation/Expander.cs:linje 1879
2>     ved Microsoft.Build.Evaluation.Expander`2.ItemExpander.ExpandExpressionCaptureIntoItems[S,T](ItemExpressionCapture expressionCapture, Expander`2 expander, IItemProvider`1 items, IItemFactory`2 itemFactory, ExpanderOptions options, Boolean includeNullEntries, Boolean& isTransformExpression, IElementLocation elementLocation) i /_/src/Build/Evaluation/Expander.cs:linje 1765
2>     ved Microsoft.Build.Evaluation.LazyItemEvaluator`4.IncludeOperation.SelectItems(Builder listBuilder, ImmutableHashSet`1 globsToIgnore) i /_/src/Build/Evaluation/LazyItemEvaluator.IncludeOperation.cs:linje 70
2>     ved Microsoft.Build.Evaluation.LazyItemEvaluator`4.LazyItemOperation.Apply(Builder listBuilder, ImmutableHashSet`1 globsToIgnore) i /_/src/Build/Evaluation/LazyItemEvaluator.LazyItemOperation.cs:linje 56
2>     ved Microsoft.Build.Evaluation.LazyItemEvaluator`4.MemoizedOperation.Apply(Builder listBuilder, ImmutableHashSet`1 globsToIgnore) i /_/src/Build/Evaluation/LazyItemEvaluator.cs:linje 190
2>     ved Microsoft.Build.Evaluation.LazyItemEvaluator`4.LazyItemList.ComputeItems(LazyItemList lazyItemList, ImmutableHashSet`1 globsToIgnore) i /_/src/Build/Evaluation/LazyItemEvaluator.cs:linje 312
2>     ved Microsoft.Build.Evaluation.LazyItemEvaluator`4.LazyItemList.GetItemData(ImmutableHashSet`1 globsToIgnore) i /_/src/Build/Evaluation/LazyItemEvaluator.cs:linje 304
2>     ved System.Linq.Enumerable.<SelectManyIterator>d__17`2.MoveNext()
2>     ved System.Linq.Buffer`1..ctor(IEnumerable`1 source)
2>     ved System.Linq.OrderedEnumerable`1.<GetEnumerator>d__1.MoveNext()
2>     ved Microsoft.Build.Evaluation.Evaluator`4.Evaluate(ILoggingService loggingService, BuildEventContext buildEventContext) i /_/src/Build/Evaluation/Evaluator.cs:linje 713
2>     ved Microsoft.Build.Evaluation.Project.Reevaluate(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext) i /_/src/Build/Definition/Project.cs:linje 2728
2>     ved Microsoft.Build.Evaluation.Project.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext) i /_/src/Build/Definition/Project.cs:linje 2688
2>     ved Microsoft.Build.Evaluation.Project.Initialize(IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext) i /_/src/Build/Definition/Project.cs:linje 2799
2>     ved Microsoft.Build.Evaluation.Project..ctor(String projectFile, IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectCollection projectCollection, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext) i /_/src/Build/Definition/Project.cs:linje 516
2>     ved Microsoft.Build.Evaluation.Project..ctor(String projectFile, IDictionary`2 globalProperties, String toolsVersion, ProjectCollection projectCollection) i /_/src/Build/Definition/Project.cs:linje 434
2>     ved Bridge.Translator.Translator.OpenProject(String location, IDictionary`2 globalProperties) i D:\Git\Bridge\Compiler\Translator\Translator\Translator.ProjectReader.cs:linje 37
2>     ved Bridge.Translator.Translator.EnsureProjectProperties() i D:\Git\Bridge\Compiler\Translator\Translator\Translator.ProjectReader.cs:linje 51
2>     ved Bridge.Translator.TranslatorProcessor.SetTranslatorProperties() i D:\Git\Bridge\Compiler\Translator\Translator\TranslatorProcessor.cs:linje 352
2>     ved Bridge.Translator.TranslatorProcessor.PreProcess() i D:\Git\Bridge\Compiler\Translator\Translator\TranslatorProcessor.cs:linje 34
2>     ved Bridge.Builder.Program.Main(String[] args) i D:\Git\Bridge\Compiler\Builder\Program.cs:linje 41
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(1328,5): error MSB3073: The command "
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(1328,5): error MSB3073: REM Run Bridge Compiler for the project
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(1328,5): error MSB3073: "D:\Git\Bridge\Compiler\Builder\bin\Debug\bridge.exe" -p "D:\Git\Bridge\Bridge\Bridge.csproj" -b "D:\Git\Bridge\Bridge\bin\Debug\Bridge.dll" -cfg "Debug" --platform "AnyCPU"
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(1328,5): error MSB3073: " exited with code 1.
3>------ Build started: Project: Bridge.Translator.Tests, Configuration: Debug Any CPU ------
3>  All packages listed in packages.config are already installed.
3>  Bridge.Translator.Tests -> D:\Git\Bridge\Compiler\TranslatorTests\bin\Debug\Bridge.Translator.Tests.dll
========== Build: 2 succeeded, 1 failed, 5 up-to-date, 0 skipped ==========
```

**Comment by fabriciomurta** [comment]
This looks like an issue introduced with the VS upgrade. If you search google, since November 2019 some people started raising this same issue with `https://github.com/OmniSharp/omnisharp-vscode/issues/3396`

There's a similar issue it seems in OmniSharp/omnisharp-vscode#3396. They disclaimed the issue, blaming a vscode-msbuild extension. Maybe you fell the same path?

**Comment by ChrML** [comment]
I will check tommorrow, @fabriciomurta

**Comment by ChrML** [comment]
@fabriciomurta Issue seems to be caused by newer versions of Visual Studio not being compatible with older versions of the *Microsoft.Build* and *Microsoft.Build.Framework* packages.

Additionally there's a new warning introduced with newer Visual Studio that prevents Bridge.NET from building.

I've created a merge request for a solution that worked for me. However I have no idea how this affects the build/release pipeline, or the tools used by existing developers. But newer is usually better... an upgrade will be required sooner or later anyway.

**Comment by joao-neves95** [comment]
A workaround I'm using is to create a new Bridge project and manually copy the Bridge.dll and Bridge.xml files to the bin of the Standard 2.1 project I'm doing.

**Comment by ChrML** [comment]
Resolved by #4118.
